scripts/tests/py/metadata.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from pathlib import Path
+import re
+import yaml
+
+
+@dataclass
+class Metadata:
+    py_script_path: str = None
+    run: str = None
+    app: str = None
+    factoryreset: bool = False
+    comissioning_method: str = None
+    discriminator: int = None
+    kvs: str = None
+    storage_path: str = None
+    on_network_commission: str = None
+    passcode: int = None
+    trace_to: str = None
+    hex_arg: str = None
```

Comment:
hex_arg/bool_arg/int_arg seem off here.

Could we start with a reduced subset and add these as needed as we start updating tests?

This is a very long list right now and I am not sure we need it all:

- int/bool/hex arg are questionalbe: should probably not be here or should have different naming
- I believe `trace_to` are hardcoded and the same for every app. We could avoid having these.